### ISSUE TITLE
Question: Is it posible that some HTTP status are not being logged?

### ISSUE THREAD

**Issue by JustDevZero** [bunkerity/bunkerized-nginx#112]
Related on the ticket I started on:

https://github.com/bunkerity/bunkerized-nginx/issues/110

Redsys is telling me that my domain is giving a HTTP 303 and they cannot continue with any operation.

Weird thing is that django is not receiving this on it's logs.

Tried to search for the 303 on nginx logs and nothing, nada.

I have also disabled mod_security just in case... but still they're receiving that 303.

If I try to do the same call via curl it works, so I guess it's somehow being blocked for the useragent or something?

Is there any way to enable DEBUG logs or something to see if that 303 is caused on nginx/mod_security/our side?

Many thanks.

**Comment by bunkerity** [comment]
Hello @JustDevZero,

Can you give us more details like compose file, variables, logs, curl command, custom config, ... ? Don't forget to edit confidential data like domains or passwords.

This is weird because we never use 303 code inside the configs and every security stuff should send a 403 or close connection.

Here are some ideas to debug : 
- Check that your callback is good on Redsys side (http vs https, trailing /, ...)
- Check that you receive a request from Redsys by looking at the logs of the container (not only the 303 code)
- Disable some security features via environment variables ([external blacklists](https://github.com/bunkerity/bunkerized-nginx#external-blacklists), [dnsbl](https://github.com/bunkerity/bunkerized-nginx#dnsbl) and [ModSecurity](https://github.com/bunkerity/bunkerized-nginx#modsecurity)) and test again
- Useful logs should be already displayed but you can also check /var/log/nginx/modsec_audit.log just in case (even if I doubt ModSecurity will send a 303...)

**Comment by JustDevZero** [comment]
Ey, thanks for thelp.  I really appreciate it.

That's on my head what makes sense, that in case of something being denied on firewall level, it should be a close connection or a 403, not a 303, 303 means redirection and if I try to reach the webpage with curl, I don't get 303 but 200. So... I don't underestand why are they complaining a 303, so... as don't provide me any headers or whatever to check I need to play "catch and hunt" to an HTTP Status that doesn't appear on logs.

So, for me, they're lying.

For them, I'm lying.

:D

So... by pieces, compose file on pastebin:

https://pastebin.com/9ZKLjLNk

I let the autoconfigure without custom conf just in case...

As for the ideas for debugging i'll try to see if it works without http, cuz in the logs there's nothing.

The only thing that comes to my mind is that they use a crappy "Java" as useragent and that's on Crawlers deny list:

https://raw.githubusercontent.com/JayBizzle/Crawler-Detect/master/raw/Crawlers.txt

But then again, that should be a 403, not a 303...  That's why I wanted asked for some "debug mode".

**Comment by JustDevZero** [comment]
@bunkerity Just a heads up, thanks for the ideas I managed to compare stuff between bunkerity nginx and "poor mans standard" nginx and I've got some good news.

I found out that non bunkerity was receiving that callback (l leave the IP because it's a public service ip so... it's fine) and this way maybe people can find this through google:

nginx_nginx.0.hwyeumhgz5aj@webserver2    | django.subdomain 195.76.9.187 - - [08/Apr/2021:20:20:27 +0000] "POST /redsys_callback HTTP/1.1" 200 0 "-" "Java/1.7.0"


So, I've checked the useragent and it's indeed on that forbidden deny list, so yeah, when I tried to curl with that useragent, you were right, we've got a "close connection", that doesn't appear on nginx log, neither mod_security, but it is blocked, something at some point blocked it.

I've managed to setup bunkerity with "everything disabled" and it worked...

So... Is there a way to add a whitelist for useragent as per an ip range?

This range should be whitelisted as they're SO ARROGANT to not add a proper normal aproved useragent.

Also, still, why is not being logged if it "closes connection"?
